### ISSUE TITLE
GULAppDelegateSwizzler crash

### ISSUE THREAD

**Issue by WillieWangWei** [firebase/firebase-ios-sdk#3859]
# Describe your environment

  * Xcode version: 11.0 (11A419c)
  * Firebase SDK version: 6.4.0
  * Firebase Component: Firebase/Core
  * Component version: 6.4.0

# Describe the problem

Using a third-party login, when I get a authorization and back to my app, a crash occurs:

![image](https://user-images.githubusercontent.com/25432264/64933390-b02da500-d877-11e9-903d-342c60b5e10f.png)

However, there is no longer a crash after setting `FirebaseAppDelegateProxyEnabled` to `NO`.

# Steps to reproduce:

Looks like it will crash when calling the method `application:openURL:options:`.

**Comment by maksymmalyhin** [comment]
@WillieWangWei Thank you for the posted issue. It looks like the `application:openURL:options:` is called recursively for some reason in your case. Could you share `mtjApplication:openURL:options:` method code to understand why it happens? If it is not possible you may try to inspect the code yourself paying attention to the following:
1. Are the conditions when method `application:openURL:options:` is called (directly or indirectly) form itself? 
2. Is there other code to swizzle the App Delegate which may work incorrectly together with `GULAppDelegateSwizzler`?

**Comment by WillieWangWei** [comment]
Thank you for your help. I solved this problem. It is because `GULAppDelegateSwizzler` and `BaiduMobStatCodeless` conflict. `BaiduMobStatCodeless` is a Baidu analysis SDK, which has many users in China. In order to use `Firebase`, I removed `BaiduMobStatCodeless`. But I have two new questions:

1. If I set `FirebaseAppDelegateProxyEnabled` to `NO`, can `Firebase` still work well?
2. The `GULAppDelegateSwizzler` swizz method seems to conflict with other code (like my case). Is there any better way to solve such general problems?
